platform/machine/azure/flight.go

```diff
@@ -146,19 +152,25 @@ func (af *flight) NewCluster(rconf *platform.RuntimeConfig) (platform.Cluster, e
 		ac.sshKey = af.FakeSSHKey
 	}
 
-	ac.ResourceGroup, err = af.api.CreateResourceGroup("kola-cluster")
-	if err != nil {
-		return nil, err
-	}
+	if af.ImageResourceGroup != "" && af.ImageStorageAccount != "" {
```

Comment:
If I understand this change correctly, the idea is that this is initialized the first time that this function is called, and after that it keeps re-using these values, is that correct? If so, could you add this to the function doc? Maybe "If no resource groups and storage accounts have been created for this flight yet, this will also create them" ?

Reply:
The function is called at least for every test, and it is not initializing something to reuse it. The function either uses the flight's Resource Group or creates a new one.
Added the documentation.

Reply:
Thanks for the extra docs, it makes more sense now! :)